analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpDiagnosticAnalyzerContextHelper.cs

```diff
@@ -68,7 +68,7 @@ public static void ReportDiagnosticIfNonGenerated(this SymbolAnalysisContext con
             }
         }
 
-        internal static bool ShouldAnalyze(this SyntaxTree tree, AnalyzerOptions options, Compilation compilation) =>
-            DiagnosticAnalyzerContextHelper.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, tree, compilation, options);
+        internal static bool ShouldAnalyze(this SyntaxTree tree, SonarAnalysisContext context, Compilation compilation, AnalyzerOptions options) =>
```

Comment:
This extension actually hid the fact, that we can use cache here. Instead, it used overload without `context` and therefore without cache